includes/api/class-wc-rest-coupons-controller.php

```diff
@@ -220,31 +219,49 @@ protected function prepare_item_for_database( $request ) {
 			}
 		}
 
-		// Code.
-		if ( ! empty( $schema['properties']['code'] ) && ! empty( $request['code'] ) ) {
-			$coupon_code = apply_filters( 'woocommerce_coupon_code', $request['code'] );
-			$id = $coupon->get_id() ? $coupon->get_id() : 0;
-
-			// Check for duplicate coupon codes.
-			$coupon_found = $wpdb->get_var( $wpdb->prepare( "
-				SELECT $wpdb->posts.ID
-				FROM $wpdb->posts
-				WHERE $wpdb->posts.post_type = 'shop_coupon'
-				AND $wpdb->posts.post_status = 'publish'
-				AND $wpdb->posts.post_title = '%s'
-				AND $wpdb->posts.ID != %s
-			 ", $coupon_code, $id ) );
-
-			if ( $coupon_found ) {
-				return new WP_Error( 'woocommerce_rest_coupon_code_already_exists', __( 'The coupon code already exists', 'woocommerce' ), array( 'status' => 400 ) );
+		// Handle all writable props
+		foreach ( $data_keys as $key ) {
+			$value = $request[ $key ];
+
+			if ( ! is_null( $value ) ) {
+				switch ( $key ) {
+					case 'code' :
+						$coupon_code = apply_filters( 'woocommerce_coupon_code', $value );
+						$id          = $coupon->get_id() ? $coupon->get_id() : 0;
+
+						// Check for duplicate coupon codes.
+						$coupon_found = $wpdb->get_var( $wpdb->prepare( "
```

Comment:
Since we use this query in a few places could we create a helper function?

We already have a `wc_get_coupon_code_by_id` so we could add a complementary `wc_get_coupon_id_by_code` function and see if the ID differs from `$id`.

`WC_Coupon` has a `get_coupon_id_from_code` function which caches its value too to avoid more queries... but it's a private method. It could be updated to call the same code, though.